### PR TITLE
 [Misc] Usage Doc for Prefill-decoding Disaggregation

### PR DESCRIPTION
**Pull request by Xinyi-ECNU**

### CODE REVIEW

File: docs/Prefill-decoding_Disaggregation.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+## Supported Features
+1. Requests can be **automatically migrated** from prefill instance to decoding instances.
+
+2. Users can specify the number of prefill instances.
+
+3. Llumnix supports both **one-to-many and many-to-one migrations** from prefill to decoding instances.
+
+4. Decoding instances can migrate requests among themselves based on different scheduling strategies (e.g. load-balance).
+
```
Comment (zhypku) [inline docs/Prefill-decoding_Disaggregation.md:18]: Please add a "Limitations" section here, and explain that this is currently an experimental feature to demonstrate the feasibility of implementing PD-D in Llumnix. And explain: we haven't done advanced performance optimization like per-layer KV cache transfer; simplified PD assignment policy.

File: docs/Prefill-decoding_Disaggregation.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+Therefore, Llumnix is inherently well-suited for the "special" cross-instance scheduling requirements in prefill-decoding disaggregation. Specifically, it limits the initial **dispatch only to the prefill instances** and the **migration between prefill and decoding instances**. 
+
+## Supported Features
+1. Requests can be **automatically migrated** from prefill instance to decoding instances.
+
+2. Users can specify the number of prefill instances.
+
+3. Llumnix supports both **one-to-many and many-to-one migrations** from prefill to decoding instances.
```
Comment (zhypku) [inline docs/Prefill-decoding_Disaggregation.md:15]: no need to mention "one-to-many and many-to-one". That's only implementation details

File: docs/Prefill-decoding_Disaggregation.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Prefill-decoding Disaggregation
```
Comment (zhypku) [inline docs/Prefill-decoding_Disaggregation.md:1]: add "experimental" to the title

**Comment by KuilongCui** [comment]
as there are some bugs for migraiton, the ci didn't pass. 
I will fix it later

File: docs/Prefill-decoding_Disaggregation.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+4. Decoding instances can migrate requests among themselves based on different scheduling strategies (e.g. load-balance).
+
+
+## How to use
+Llumnix only uses two arguments to enable prefill-decoding disaggregation for simplicity. 
+- `--enable-pd-disagg True` is used to enable prefill-decoding disaggregation.
+- `--num-available-dispatch-instances` is used to configure the number of prefill instances. 
+
+Note that `num-available-dispatch-instances` < `initial_instance-num` especially when `--enable-scaling` is not set, as it determines the number of decoding instances.
```
Comment (KuilongCui) [inline docs/Prefill-decoding_Disaggregation.md:25]: If the available instances become less than or equal to num-available-dispatch-instances, for example, some CUDA error causes some instances to crash, what will happen?

**Comment by KuilongCui** [comment]
Missing a description of the failover mechanism.

File: docs/Prefill-decoding_Disaggregation.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+5. Fine tuning of the scheduling policies.
+
+We are actively working on these items. Stay tuned :)
+
+## How to use
+Llumnix uses two simple arguments to enable prefill-decoding disaggregation in the current version. 
+- `--enable-pd-disagg True` is used to enable prefill-decoding disaggregation.
+- `--num-available-dispatch-instances` is used to configure the initial number of prefill instances. 
+
+Note that one should make sure that `num-available-dispatch-instances` is smaller than `initial_instance-num` (especially when `--enable-scaling` is not set), otherwise there would be no instances for decoding.
```
Comment (Xinyi-ECNU) [inline docs/Prefill-decoding_Disaggregation.md:49]: my typo... `initial_instance-num` -> `initial-instances`

Comment (zhypku) [reply docs/Prefill-decoding_Disaggregation.md]: fixed

**Review by zhypku — APPROVED** [review]

File: docs/Prefill-decoding_Disaggregation.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
+# Prefill-decoding Disaggregation (Experimental)
+
+Prefill-decoding disaggregation is a technique that computes the prefill and decoding phases on separate instances, designed mainly for reducing the inteference between the two phases and better utilizing heterogeneous hardware. For each request, following the prefill phase, the system migrates the generated key-value (KV) cache to the decoding instance and continues the computation. 
+
+We find Llumnix well-suited for implementing P-D disaggregation, because this technique is inherently a special request scheduling policy and fits well in Llumnix's modeling for request scheduling. Specifically, P-D disaggregation can be decomposed into two rules (shown below): (1) a special dispatching rule, i.e., P-instances-only; and (2) a special migration rule, i.e., migrate to D instances after one step. Llumnix provides an implementation of P-D disaggregation following this principle.
+
+<div align=center>
+<img src="./pdd_1.png" align="center" width=80%/>
```
Comment (KuilongCui) [inline docs/Prefill-decoding_Disaggregation.md:8]: Update the figure file name to reflect the content of the image for pdd_1.png and pdd_2.png

**Comment by KuilongCui** [comment]
LGTM ：）

**Review by KuilongCui — APPROVED** [review]

File: docs/Prefill-decoding_Disaggregation.md (hunk 6 above)
Comment (zhypku) [reply docs/Prefill-decoding_Disaggregation.md]: done

**Pull request merged**